src/Lookup/Lookup.tsx

```diff
@@ -456,8 +460,9 @@ export default class Lookup extends React.PureComponent<LookupProps, State> {
 	onMenuScrollToBottom = async () => {
 		const { search, optionsCache } = this.state;
 		const currentOptions = optionsCache[search];
+		const pageSize = this.props.pageSize ?? 20;
 
-		if (currentOptions) {
+		if (currentOptions && currentOptions.options.length % pageSize === 0) {
```

Comment:
This is not bulletproof as pageSize is dynamic prop and in theory can change during scrolling. More correct behavior would be to set a flag in state when no more records are available (number of items returned < number of items requested per page). But that solution will require more wiring in some other places.

Reply:
Sure I can implement the solution, but I can't really see that the pageSize is going to change during runtime?